packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1421,18 +1436,23 @@ export class GarbageCollector implements IGarbageCollector {
 	 * @param gcData - The data representing the reference graph on which GC is run.
 	 * @param gcResult - The result of the GC run on the gcData.
 	 * @param currentReferenceTimestampMs - The timestamp to be used for unreferenced nodes' timestamp.
+	 * @returns - A list of sweep ready nodes. (Nodes ready to be deleted)
 	 */
-	private updateCurrentState(
+	private updateMarkPhase(
 		gcData: IGarbageCollectionData,
 		gcResult: IGCResult,
 		currentReferenceTimestampMs: number,
+		logger: ITelemetryLogger,
 	) {
-		this.gcDataFromLastRun = cloneGCData(gcData);
-		this.tombstones = [];
+		// Update the state since the last GC run. There can be nodes that were referenced between the last and
```

Comment:
This comment doesn't explain the whole picture. Basically, what it should say is that we get references from the current GC run + references between previous and current run and then update their state.